crash: nil pointer dereference in resourceScalewayRdbDatabaseRead

Doing a `terraform destroy` but I've destroyed the rdb instance manually on the web console.

The crash is now systematic : I can't run `terraform destroy` anymore.

```
Stack trace from the terraform-provider-scaleway_v2.9.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe8f57e]

goroutine 112 [running]:
github.com/scaleway/terraform-provider-scaleway/v2/scaleway.resourceScalewayRdbDatabaseRead({0x1414f48, 0xc0002d0780}, 0xc00086b380, {0xf254e0?, 0xc00059d110?})
	github.com/scaleway/terraform-provider-scaleway/v2/scaleway/resource_rdb_database.go:165 +0x2be
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0007176c0, {0x1414f80, 0xc000840690}, 0xd?, {0xf254e0, 0xc00059d110})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:724 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0007176c0, {0x1414f80, 0xc000840690}, 0xc0005692b0, {0xf254e0, 0xc00059d110})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0003f4bd0, {0x1414f80?, 0xc000840570?}, 0xc00090e700)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:613 +0x4a5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00066e280, {0x1414f80?, 0xc0005f1e60?}, 0xc0005bf2c0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:748 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x1087460?, 0xc00066e280}, {0x1414f80, 0xc0005f1e60}, 0xc0007c0690, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00026a3c0, {0x14192e0, 0xc000007a00}, 0xc0002f87e0, 0xc0005904e0, 0x1ae8b70, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc00026a3c0, {0x14192e0, 0xc000007a00}, 0xc0002f87e0, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

Error: The terraform-provider-scaleway_v2.9.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

Operation failed: failed running terraform plan (exit 1)

```

--- a/scaleway/resource_rdb_database.go
+++ b/scaleway/resource_rdb_database.go
@@ -133,9 +133,6 @@ func getDatabase(ctx context.Context, api *rdb.API, r scw.Region, instanceID, db
 		Name:       &dbName,
 	}, scw.WithContext(ctx))
 	if err != nil {
-		if is404Error(err) {
-			return nil, nil
-		}
 		return nil, err
 	}
 

--- a/scaleway/resource_rdb_privilege.go
+++ b/scaleway/resource_rdb_privilege.go
@@ -125,6 +125,10 @@ func resourceScalewayRdbPrivilegeRead(ctx context.Context, d *schema.ResourceDat
 
 	_, err = waitForRDBInstance(ctx, rdbAPI, region, instanceID, d.Timeout(schema.TimeoutRead))
 	if err != nil {
+		if is404Error(err) {
+			d.SetId("")
+			return nil
+		}
 		return diag.FromErr(err)
 	}
 
@@ -134,6 +138,10 @@ func resourceScalewayRdbPrivilegeRead(ctx context.Context, d *schema.ResourceDat
 		Name:       &userName,
 	}, scw.WithContext(ctx))
 	if err != nil {
+		if is404Error(err) {
+			d.SetId("")
+			return nil
+		}
 		return diag.FromErr(err)
 	}
 

--- a/scaleway/resource_rdb_user.go
+++ b/scaleway/resource_rdb_user.go
@@ -118,6 +118,10 @@ func resourceScalewayRdbUserRead(ctx context.Context, d *schema.ResourceData, me
 
 	_, err = waitForRDBInstance(ctx, rdbAPI, region, instanceID, d.Timeout(schema.TimeoutRead))
 	if err != nil {
+		if is404Error(err) {
+			d.SetId("")
+			return nil
+		}
 		return diag.FromErr(err)
 	}
 

